dbt_platform_helper/domain/pipelines.py

```diff
@@ -63,147 +72,67 @@ def generate(self, terraform_platform_modules_version, deploy_branch):
 
         self._clean_pipeline_config(copilot_pipelines_dir)
 
+        terraform_platform_modules_version = get_required_terraform_platform_modules_version(
+            cli_terraform_platform_modules_version,
+            platform_config_terraform_modules_default_version,
+        )
+
         if has_environment_pipelines:
-            environment_pipelines = pipeline_config[ENVIRONMENT_PIPELINES_KEY]
+            environment_pipelines = platform_config[ENVIRONMENT_PIPELINES_KEY]
+            accounts = {
+                config.get("account")
+                for config in environment_pipelines.values()
+                if "account" in config
+            }
 
-            for config in environment_pipelines.values():
-                aws_account = config.get("account")
+            for account in accounts:
                 self._generate_terraform_environment_pipeline_manifest(
-                    pipeline_config["application"],
-                    aws_account,
+                    platform_config["application"],
+                    account,
                     terraform_platform_modules_version,
-                    platform_config_terraform_modules_default_version,
                     deploy_branch,
                 )
 
         if has_codebase_pipelines:
-            account_id, _ = get_account_details()
-
-            for codebase in pipeline_config[CODEBASE_PIPELINES_KEY]:
-                self._generate_codebase_pipeline(
-                    account_id,
-                    app_name,
-                    codestar_connection_arn,
-                    git_repo,
-                    codebase,
-                    base_path,
-                    copilot_pipelines_dir,
-                    templates,
-                )
+            codebase_pipelines = platform_config[CODEBASE_PIPELINES_KEY]
+            provisioned_ecrs = set(self.ecr_provider.get_ecr_repo_names())
+            required_ecrs = {
+                codebase: f"{platform_config['application']}/{codebase}"
+                for codebase in codebase_pipelines.keys()
+                if f"{platform_config['application']}/{codebase}" in provisioned_ecrs
```

Comment:
Does this line mean that `required_ecrs` will only ever contain `provisioned_ecrs`?

Reply:
Looks like I refactored and left that `in provisioned_ecrs` filter in there twice.